src/hooks/useFormValidation.ts

```
@@ -2,274 +1,72 @@
+  [key: string]: string
 }
 
-interface FormState<T> {
-  data: Partial<T>
-  errors: Record<string, string>
-  isSubmitting: boolean
-  isValid: boolean
-  isDirty: boolean
+interface UseFormValidationReturn {
+  errors: ValidationError
+  validateField: (field: string, value: any, schema: z.ZodSchema) => string | null
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)